packages/story-editor/src/components/floatingMenu/elements/textColor.js

```diff
@@ -17,20 +17,51 @@
 /**
  * External dependencies
  */
-import { Icons } from '@googleforcreators/design-system';
 import { __ } from '@googleforcreators/i18n';
+import { useCallback } from '@googleforcreators/react';
 
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import useRichTextFormatting from '../../panels/design/textStyle/useRichTextFormatting';
+import updateProperties from '../../inspector/design/updateProperties';
+import { useStory } from '../../../app';
+import { Color, useProperties } from './shared';
 
 function TextColor() {
+  const { content, type } = useProperties(['content', 'type']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  const pushUpdate = useCallback(
+    (update) => {
+      updateSelectedElements({
+        properties: (element) => {
+          const updates = updateProperties(element, update, true);
+          return {
+            ...element,
+            ...updates,
+          };
+        },
+      });
+    },
+    [updateSelectedElements]
+  );
+  const {
+    textInfo: { color },
+    handlers: { handleSetColor },
+  } = useRichTextFormatting([{ content, type }], pushUpdate);
+
   return (
-    <IconButton
-      Icon={Icons.Pipette}
-      title={__('Change text color', 'web-stories')}
-      onClick={() => {}}
+    <Color
+      label={__('Text color', 'web-stories')}
+      value={color}
+      allowsSavedColors
+      onChange={handleSetColor}
+      hasInputs={false}
+      hasEyeDropper
+      allowsOpacity={false}
     />
```

Comment:
I feel we should incorporate the warning about bad contrast into the design menu as well. Either directly in the color picker or as a warning next to the color input in the design menu. Because when we move the design panel to a (mostly hidden) tab, users won't see it anymore.

![Screen Shot 2022-03-08 at 15 31 35](https://user-images.githubusercontent.com/637548/157320013-27bc46f7-013f-41d4-b0a1-d0dc8a7c5a59.png)

But that's probably a follow-up ticket needing UX lead.

Reply:
Added issue: https://github.com/GoogleForCreators/web-stories-wp/issues/10868